keanu-project/src/test/java/io/improbable/keanu/e2e/regression/LinearRegressionTest.java

```diff
@@ -164,7 +164,7 @@ public void youCanChooseSamplingInsteadOfGradientOptimization() {
             .withSampling(sampling)
             .build();
 
-        NetworkSamples networkSamples = sampling.getNetworkSamples().drop(samplingCount - 10000).downSample(100);
+        NetworkSamples networkSamples = sampling.getNetworkSamples().drop(samplingCount / 10).downSample(2);
```

Comment:
We've reduced the ratio of burn-in samples to samples by a huge amount here. We were initially dropping 2/3rds of our samples now we drop a tenth. @migwellian does this need to be the same as before? Same for: `LinearRidgeRegressionTest.java`

Reply:
Good question - this was sort of deliberate though as the burn in before seemed pretty high (especially as there's just two latents here and the posterior should be pretty simple)?

Also - if the test still passes, but the sampler is producing bad samples, then I think the test itself is probably flawed?

Reply:
When you turn up logging you get some stats, and they look good:

Before:

>2018-12-10 10:04:42 INFO  LinearRegressionTestUtils:122 - Gradient from sampling:     2.9360 ~ 0.0718
>2018-12-10 10:04:42 INFO  LinearRegressionTestUtils:123 - Gradient from initial data: 2.9371 ~ 0.0683
>2018-12-10 10:04:42 INFO  LinearRegressionTestUtils:124 - Intercept from sampling:     20.6597 ~ 0.4439
>2018-12-10 10:04:42 INFO  LinearRegressionTestUtils:125 - Intercept from initial data: 20.6774 ~ 0.4024

After:

>2018-12-10 10:05:41 INFO  LinearRegressionTestUtils:122 - Gradient from sampling:     2.9448 ~ 0.0727
>2018-12-10 10:05:41 INFO  LinearRegressionTestUtils:123 - Gradient from initial data: 2.9371 ~ 0.0683
>2018-12-10 10:05:41 INFO  LinearRegressionTestUtils:124 - Intercept from sampling:     20.6285 ~ 0.4234
>2018-12-10 10:05:41 INFO  LinearRegressionTestUtils:125 - Intercept from initial data: 20.6774 ~ 0.4024

tbh I'm surprised it worked as I agonised over tweaking these parameters for ages! In particular we thought that the high down-sampling rate would be necessary since if you get a lot of rejected samples then it keeps on sampling the same point and so the variance drops.

Reply:
The acceptance rates are pretty high: 0.67 for the intercept and 0.19 for the weights.

Reply:
For this graph won't we only have two latents and thus following the "basic rule" of downsampling by N would mean we're "correct"?

It seems that the new values are still giving us decent values though so the tests are probably ok as is?